File: app/lib/features/lead_image/domain/constants/image_constants.dart
class ImageConstants {
  static const int maxImageSizeBytes = 5242880; // 5MB
  static const int maxImagesPerLead = 10;
  static const List<String> supportedFormats = ['jpg', 'jpeg', 'png', 'gif', 'webp'];
  static const String base64ImagePrefix = 'data:image/';
  static const int thumbnailSize = 150;
  static const int compressionQuality = 85;
  static const int maxCacheSizeMB = 50;
  static const int compressionThreshold = 1048576; // 1MB - compress if larger than this

  const ImageConstants._();
}

File: backend/Domain/Image/Constants/ImageConstants.cs
namespace Domain.Image.Constants;

public static class ImageConstants
{
    public const int MAX_IMAGE_SIZE_BYTES = 5242880; // 5MB
    public const int MAX_FILE_NAME_LENGTH = 255;
    public const int MIN_BASE64_LENGTH = 100; // Minimum reasonable Base64 string
    public const string DEFAULT_CONTENT_TYPE = "image/jpeg";

    public static readonly string[] ALLOWED_CONTENT_TYPES =
    {
        "image/jpeg",
        "image/jpg",
        "image/png",
        "image/gif",
        "image/webp"
    };

    public static readonly string[] ALLOWED_EXTENSIONS =
    {
        ".jpg",
        ".jpeg",
        ".png",
        ".gif",
        ".webp"
    };
}